packages/kbn-interpreter/src/server/require_with_globals.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Fs from 'fs';
+import Module from 'module';
+import Vm from 'vm';
+import Path from 'path';
+
+const stripBOM = string => {
+  if (string.charCodeAt(0) === 0xFEFF) {
+    string = string.slice(1);
+  }
+  return string;
+};
+
+export function requireWithGlobals(path, extraGlobals) {
+  const mainModule = new Module(path);
+  const mainRequire = Module.createRequireFromPath(path);
+
+  // module cache for modules created in this context
+  mainRequire._cache = {};
+
+  // mimic the default node.js extension support
+  mainRequire._extensions = {
+    '.js': function (module, path) {
```

Comment:
In the spirit of not automatically providing any "magic" build tooling I did not re-implement support for babel/typescript transpilation here. Just like they do for the browser plugins will need to build their server code to run natively in node.js, as we are now doing in `x-pack/plugins/canvas/tasks/helpers/webpack.plugins.js`

Reply:
BTW, @spalger and I talked about this. This change is, if anything, more inline with how we want to see things done in Canvas and on the new platform. We want to provide a simple, predictable, dependable environment with as few moving parts as possible. With this change, if Kibana changes how it transpiles code, it won't affect plugin authors.